### ISSUE TITLE
Support Web Crypto API

### ISSUE THREAD

**Issue by baryon** [NativeScript/ios-runtime#1045]
In the NS runtime, the code will return undefine
`
var crypto = global.crypto || global.msCrypto
`
Any plan to support Web Crypto API?

https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API

**Comment by baryon** [comment]
no crypto supporting, we can’t use some npm package and features like random and hash functions directly and safety

**Comment by danielgek** [comment]
@mbektchiev honestly i think this is a must have, big companies and governments usually require a custom implementation because the framework doesn't support it OOTB, @baryon you can checkout [this repo ](https://github.com/tomvardasca/nativescript-crypto), but i don't know if it works in its current state, i have used it two years ago.

**Comment by mbektchiev** [comment]
@danielgek @baryon It is not entirely correct to state that there's no cryptographic implementation out-of-the-box because actually all crypto APIs provided by the iOS and Android OSes are directly accessible from NativeScript. For example all [iOS Security Framework's APIs](https://developer.apple.com/documentation/security?language=objc) can be freely used in {N} applications.

**Comment by danielgek** [comment]
I totally agree with you, it's there and might be not that hard to implement but requires more work than expected ;)

**Comment by mbektchiev** [comment]
Yep, that's true. Maybe having a plugin similar to [nativescript-crypto](https://github.com/tomvardasca/nativescript-crypto) which provides an API similar to Web Crypto's would be a really nice thing to have.
